scripts

```diff
@@ -0,0 +1,62 @@
+package main
+
+import "fmt"
+
+func findNumber(collection []int, n int) bool {
+
+	for _, i := range collection {
+		if i == n {
+			return true
+		}
+	}
+
+	return false
+}
+
+func findNumber1(collection []int, n int) bool {
+	index := 0
+	for index < len(collection) {
+		if collection[index] == n {
+			return true
+		}
+		index++
+	}
+
+	return false
+}
+
+func findPosition(collection []int, n int) (bool, int) {
+
+	for i := len(collection) - 1; i >= 0; i-- {
+		if collection[i] == n {
+			fmt.Printf("the position of %d is %d\n", n, i)
+			return true, i
```

Comment:
The answer to this question is it depends:

1- You decide what your function will return, in languages like Go you can return tuples that represent multiple things, in other languages you can only return one single type, meaning either you return a boolean or you return an integer. There is not right or wrong in this case, but it strictly depends on what you want to accomplish. If you want to simplify and do it the old school way you would typically return an integer which either represents the position meaning such number will be (lets called x) x >= 0 && x < length(collection) or it will be -1 which means the number was not in the collection in that sense you get both the actual position and the boolean value automatically cause you agreed that if your function returns -1 the number is not in there, in any other case it is